Automattic/jetpack
# Backup: Update to use tracking check from connection package

The idea is to refactor the tracking that we currently have in the backup package to use the one on the connection package that also validates if the site agreed the term and conditions. This logic is already implemented in [MyJetpack](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/my-jetpack/src/class-initializer.php#L187), [VideoPress](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/class-admin-ui.php#L147) and [Social](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/social/src/class-jetpack-social.php).

## Proposed changes:

* Switch to use tracking check from connection package

### Other information:


## Jetpack product discussion


1202586245449203-as-1203758786856450/f

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Spin up a new JN site with Jetpack VaultPress Backup plugin using the branch `update/backup-tracking-connection-package`. Don't add a backup plan yet.
* Open the Developer Console > Network and click on Preserve log checkbox.
* Without connecting Jetpack, visit the Jetpack VaultPress Backup plugin admin page.
* Ensure the `t.gif` is not loading.
* Now, click on `Get VaultPress Backup` to purchase a plan.
* After purchasing the plan (and redirected back to the plugin), validate the `t.gif` is now loaded and recording events.
